### PR TITLE
feat(newsletter): add initial ngrx features for @daffodil/newsletter

### PR DESCRIPTION
**Pull request by Nolan-Arnold**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/newsletter/src/actions/newsletter.actions.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Action } from '@ngrx/store';
+
+import { DaffNewsletter } from '../models/newsletter.model';
+
+export enum DaffNewsletterActionTypes {
+  NewsletterSubscribeAction = "[Daff-Newsletter] Newsletter Subscribe Action",
+  NewsletterCancelAction = "[Daff-Newsletter] Newsletter Cancel Action",
+  NewsletterSuccessSubscribeAction = "[Daff-Newsletter] Succeeded on Newsletter Subscribe Action",
+  NewsletterFailedSubscribeAction = "[Daff-Newsletter] Failed on Newsletter Subscribe Action",
+  NewsletterRetry = "[Daff-Newsletter] Retrying to cancel newsletter",
```
Comment (damienwebdev) [inline libs/newsletter/src/actions/newsletter.actions.ts]: Clarify this action text: `[Daff-Newsletter] Retrying to cancel newsletter` it should probably be:

```
[Daff-Newsletter] Retry submission
```

File: libs/newsletter/src/models/newsletter.model.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface DaffNewsletter {
```
Comment (damienwebdev) [inline libs/newsletter/src/models/newsletter.model.ts]: Probably should be called `DaffNewsletterSubmission`

Comment (Nolan-Arnold) [reply libs/newsletter/src/models/newsletter.model.ts]: Okay, this was updated through the library.

File: libs/newsletter/src/actions/newsletter.actions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nolan-Arnold) [reply libs/newsletter/src/actions/newsletter.actions.ts]: This was a typo, it was suppose to be retry submit. I addressed this in the last commit.

**Comment by lderrickable** [comment]
lgtm

**Pull request merged**